### ISSUE TITLE
Map the geometry onto a sphere

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#647]
This would make far away terrain disappear beneath the horizon instead of fading out into the fog.

**Comment by ITR13** [comment]
Normally there's a trick with vertex shaders that can be done to "bend" the world, but that usually requires higher vertex counts to look good, so not sure how'd that look with good meshing.

**Comment by IntegratedQuantum** [comment]
You don't actually need high vertex counts, since the effect should be fairly small, since it should be about the same magnitude as the curvature of the earth.
As an optimization it might make sense to consider only applying the expensive (it requires a sqrt) effect at the chunk corners, with linear interpolation in between.

**Comment by IntegratedQuantum** [comment]
This only makes sense for height-map based terrain #173 where there is no terrain and a large enough range.

**Comment by IntegratedQuantum** [comment]
> Instead of making everything round, only chunks at the edge of the render distance will have wrapping.
This will help it from feeling weird, like how it was during previous testing.

Either way, it must not be applied to regular chunks to avoid rendering artifacts from incorrectly sorted transparent blocks, that's why we'd need #173 which would not contain transparency.

**Comment by TheNewSector** [comment]
or
make the world a sphere (like seed of andromeda)
-completely unrealistic and bad idea (must add either distortion or non-euclidian corners, not infinite, requires rewriting the entire renderer, world gen, etc...) -would probably be an option in the word generation screen
would be super cool though

**Comment by IntegratedQuantum** [comment]
> make the world a sphere (like seed of andromeda)

No, it's not worth the effort. Especially since you couldn't even tell the difference in normal gameplay.

**Comment by Argmaster** [comment]
I think this does not fit the style of the game - there are literally no curves anywhere and in my opinion it should stay this way - this is voxel pixel art game, we should cherish our Euclidean space and straight lines.

**Comment by IntegratedQuantum** [comment]
I still think it would look nice to have just a little curvature on the large scale of the world.
